src/commands/runDraftTool/runDraftDeployment.ts

```diff
@@ -128,27 +128,56 @@ async function multiStepInput(
     input: MultiStepInput,
     state: Partial<State>,
     step: number
-  ) {
+  ): Promise<(input: MultiStepInput) => any> {
     const namespaces: k8s.V1Namespace[] = await listNamespaces();
     const items = namespaces.map((namespace) => {
       return {
         label: `${namespace.metadata?.name}`,
-        description: namespace.metadata?.name,
       };
     });
+    const newNamespace = "New Namespace";
+    const itemsWithNew: vscode.QuickPickItem[] = [
+      { label: newNamespace },
+      ...items,
+    ];
+
     const pick = await input.showQuickPick({
       title,
       step: step,
       totalSteps: totalSteps,
-      placeholder: 'Kubernetes namespace (e.g, myapp)',
-      items: items,
-      activeItem: typeof state.namespace !== 'string' ? state.namespace : undefined,
-      shouldResume: shouldResume
+      placeholder: "Kubernetes namespace",
+      items: itemsWithNew,
+      activeItem: undefined,
+      shouldResume: shouldResume,
     });
+
+    if (pick.label === newNamespace) {
+      return (input: MultiStepInput) => inputNewNamespace(input, state, step);
+    }
+
     state.namespace = pick.label;
     return (input: MultiStepInput) => inputName(input, state, step + 1);
   }
 
+  async function inputNewNamespace(
+    input: MultiStepInput,
+    state: Partial<State>,
+    step: number
+  ) {
+    const ns = await input.showInputBox({
+      title,
+      step: step,
+      totalSteps: totalSteps,
+      validate: async () => undefined,
+      value: "",
+      prompt: "Namespace",
+      shouldResume: shouldResume,
+    });
+    await createNamespace(ns);
+    state.namespace = ns;
+    return (input: MultiStepInput) => selectNamespace(input, state, step);
```

Comment:
doesn't this cause circular loop? select namespace --> input new namespace --> select namespace ? should it be inputName method ? Please correct me if my understanding is wrong.

Reply:
That's intentional for now. We don't have proper error checking on the create new namespace so if it fails then later operations will fail. By looping back to the select after creating the namespace we guarantee the namespace actually exists.

Reply:
Since this is a POC we are okay doing this. But for production we need revisit all these :)